packet.go

```diff
@@ -292,6 +370,10 @@ func (p *Packet) decodeIp6(recur int) {
 	if p.ip6hdr == nil {
 		p.ip6hdr = new(Ip6hdr)
 	}
+	if recur > 0 {
+		p.EncapType = IP_IN_IP
```

Comment:
Same comment as decodeIP().

Reply:
ip-ip and gre both doesn't have mac. gre uses EthProto to identify next protocol. It is same Ethproto used by ethernet header to identify next protocol.